src/Simulation/Simulators.Impl/QuantumSimulator/SWAP.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Quantum.Simulation.Core;
+
+namespace Microsoft.Quantum.Simulation.Simulators
+{
+
+    public partial class QuantumSimulator
+    {
+        public class QSimSWAP : Intrinsic.SWAP
+        {
+            [DllImport(QSIM_DLL_NAME, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl, EntryPoint = "MCX")]
+            private static extern void MCX(uint id, uint count, uint[] ctrls, uint qubit);
+
+            private QuantumSimulator Simulator { get; }
+
+            public QSimSWAP(QuantumSimulator m) : base(m)
+            {
+                this.Simulator = m;
+            }
+
+            public override Func<(Qubit, Qubit), QVoid> __Body__ => (args) =>
+            {
+                var (qubit1, qubit2) = args;
+                var ctrls = new QArray<Qubit>(qubit1);
+                Simulator.CheckQubits(ctrls, qubit2);
+
+                MCX(Simulator.Id, (uint)ctrls.Length, ctrls.GetIds(), (uint)qubit2.Id);
```

Comment:
This is a bug in the native SWAP support that is fixed as part of #383, which will need to be retargeted as a PR against main once this PR is merged.

Reply:
To ensure that tests passed as expected, I had to pull in this portion of #383, so SWAP is now be fixed.